src/datasets/arrow_dataset.py

```diff
@@ -3738,33 +3727,33 @@ def _select_with_indices_mapping(
     @fingerprint_transform(inplace=False, ignore_kwargs=["load_from_cache_file", "indices_cache_file_name"])
     def sort(
         self,
-        column: str,
-        reverse: bool = False,
-        kind: str = None,
-        null_placement: str = "last",
+        column_names: Union[str, Sequence_[str]],
+        reverse: Union[bool, Sequence_[bool]] = False,
+        null_placement: str = "at_end",
         keep_in_memory: bool = False,
         load_from_cache_file: bool = True,
         indices_cache_file_name: Optional[str] = None,
         writer_batch_size: Optional[int] = 1000,
         new_fingerprint: Optional[str] = None,
+        kind="deprecated",
     ) -> "Dataset":
-        """Create a new dataset sorted according to a column.
+        """Create a new dataset sorted according to a single or multiple columns.
+        The transformation is applied to all the datasets of the dataset dictionary.
 
-        Currently sorting according to a column name uses pandas sorting algorithm under the hood.
-        The column should thus be a pandas compatible type (in particular not a nested type).
+        Currently sorting according to a column name(s) uses pyarrow's sort_indices sorting algorithm under the hood.
+        The column(s) should thus be a pyarrow compatible type (in particular not a nested type).
```

Comment:
I don't think this info is particularly useful, so feel free to remove it (`ArrowNotImplementedError` is thrown in that case anyways)

```suggestion
```